fix: strengthen the authentication theorems of ISO-DH

TWal opened the pull request:
I noticed that in the authentication theorems of ISO-DH we only need to talk about corruption of signature keys of the other party, the attacker cannot use corrupted DH private keys to break authentication (unlike NSL where corrupting a nonce can be used to break authentication).

fabian-hk commented:
This looks good. One thing that is unrelated to this PR: Aren't the authentication lemmas named the wrong way? So `initiator_authentication` should be `responder_authentication` and `responder_authentication` should be `initiator_authentication`

TWal commented:
Ah, correct!

I changed that (renaming theorems seems close enough to strengthening them to do it in the same PR), and also updated the "explanation asserts" in the security theorem proofs.
